omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```diff
@@ -0,0 +1,266 @@
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.util.OpenmrsClassLoader;
+
+import java.util.*;
+
+@Resource(name = RestConstants.VERSION_1 + "/tag", supportedClass = Tag.class, supportedOpenmrsVersions = { "1.11.*",
+        "1.12.*", "2.0.*", "2.1.*" })
+public class TagResource extends DelegatingCrudResource<Tag> {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#newDelegate()
+	 */
+	@Override
+	public Tag newDelegate() {
+		return new Tag();
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getByUniqueId(String)
+	 */
+	@Override
+	public Tag getByUniqueId(String uuid) {
+		
+		try {
+			UUID uuid1 = UUID.fromString(uuid);
+			return Context.getService(TagService.class).getTagByUuid(uuid);
+		}
+		catch (IllegalArgumentException e) {
+			Tag tag = new Tag();
+			tag.setTag(uuid);
+			return tag;
+		}
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#save(Object)
+	 */
+	@Override
+	public Tag save(Tag tag) {
+		return getService().saveTag(tag);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#delete(Object,String, RequestContext)
+	 */
+	@Override
+	protected void delete(Tag tag, String reason, RequestContext context) {
+		throw new ResourceDoesNotSupportOperationException("delete not allowed on tag");
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#purge(Object, RequestContext)
+	 */
+	@Override
+	public void purge(Tag tag, RequestContext context) {
+		getService().purgeTag(tag);
+	}
+	
+	/**
+	 * Returns the DefaultRepresentation
+	 */
+	@RepHandler(DefaultRepresentation.class)
+	public SimpleObject asDef(Tag delegate) throws ConversionException, ClassNotFoundException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("tag");
+		if (StringUtils.isNotBlank(delegate.getObjectType())) {
+			description.addProperty("uuid");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+		}
+		List<Tag> tagList = Context.getService(TagService.class).getTags(delegate.getTag(), true);
+		Iterator<Tag> tagIterator = tagList.iterator();
+		while (tagIterator.hasNext()) {
+			Tag iterator = tagIterator.next();
+			description.addLink(getResourceName(iterator.getObjectType()),
+			    getUriOfObject(iterator.getObjectType(), iterator.getObjectUuid()));
+		}
+		description.addSelfLink();
+		description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		if (tagList.size() == 0) {
+			description = null;
```

Comment:
there can be a case where we ask for tag/my-favourites, and there is no such tag in the database. however according to the logic a new Tag object will be created and sent for its representation. So this is a final check, wherein if no such tag actually exists in the database the description returns null